src/Symfony/Component/Security/Core/Authorization/TraceableAccessDecisionManager.php

```diff
@@ -37,10 +37,14 @@ public function __construct(AccessDecisionManagerInterface $manager)
         $this->manager = $manager;
 
         // The strategy and voters are stored in a private properties of the decorated service
-        $reflection = new \ReflectionProperty(\get_class($manager), 'strategy');
-        $this->strategy = $reflection->getValue($manager);
-        $reflection = new \ReflectionProperty(\get_class($manager), 'voters');
-        $this->voters = $reflection->getValue($manager);
+        if ((new \ReflectionObject($manager))->hasProperty('strategy')) {
```

Comment:
```suggestion
        if (property_exists($manager, 'strategy')) {
```

same below, should be more efficient than reflection

Reply:
perfect suggestion